app/models/notifications/mentioned_in_comment.rb

```diff
@@ -10,8 +10,15 @@ def deleted_translation_key
       "notifications.mentioned_in_comment_deleted"
     end
 
-    def self.filter_mentions(mentions, _mentionable, _recipient_user_ids)
-      mentions
+    def self.filter_mentions(mentions, mentionable, _recipient_user_ids)
+      ppl = mentionable.people_allowed_to_be_mentioned
+      if ppl == "all"
+        mentions
+      else
+        mentions.to_ary.delete_if {|mention|
```

Comment:
> "features first, performance second"

That shouldn't be an excuse to ignore performance ;)

But I think in this case the `where` is ok for the beginning, it shouldn't cause any issues, and if it does, we can change this in the future. This is also only in background processing (sidekiq), so no slow frontend actions because of this. But with 10000 it isn't fast anymore, compared to 0.x ms with 10, but I think we can live with it, it doesn't kill the DB.

> Well, the default scope is "All aspects"

That still doesn't mean that all contacts are participating.